sycl/test/CMakeLists.txt

```diff
@@ -60,7 +60,7 @@ add_lit_testsuite(check-sycl-spirv "Running device-agnostic SYCL regression test
   )
 
 add_custom_target(check-sycl)
-add_dependencies(check-sycl check-sycl-spirv)
+add_dependencies(check-sycl check-sycl-spirv check-libclc)
```

Comment:
I'm not sure it works. `libclc` is not enabled by default (only with CUDA/HIP back-ends), so I assume there might be problems if this library is not built. Right?

I suggest adding a separate command to this GitHub Actions script: https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl_linux_build_and_test.yml#L136.
@alexbatashev can help with that.

Reply:
You are right: https://github.com/jchlanda/llvm/blob/jakub/check_sycl_check_libclc/buildbot/configure.py#L62
Do you think it's worth adding it to the list of `llvm_enabled_projects` by default?

Reply:
No, it's a dead code by default, but it must be enabled in pre-/post-commit CI validation.
It might be worth adding when CUDA or HIP backends included to the build, but I'd like to avoid the drawback mentioned in the description:
> The only slight drawback is that if check-libclc fails it will not proceed to running check-sycl.

To avoid this, we can run `check-libclc` separately from `check-sycl` as we do for the compiler (i.e. `check-clang`). 
If we need simplify the workflow for DPC++ compiler developers we can make `check-sycl-toolchain` CMake target with dependencies on `check-*` testing the compiler, runtime and other DPC++ components, which can be executed concurrently and do not block each other. Python script instead of cmake target works as well.

Reply:
@jchlanda Alexey is right, you don't need to add dependencies to the `check-sycl`. Please, adjust `configure.py` call [here](https://github.com/intel/llvm/blob/f058fe076fc763150ceb1fbb24834b3fabf2ebc3/.github/workflows/sycl_linux_build_and_test.yml#L111) and just add a new step for the build job as @bader described earlier. All the logs will be public, both pre-commit and post-commit testing will be enabled, and you'll be able to modify the run command as you see fit.

Reply:
I've folded it into: b8f99bc540d190629ce24068bdde3f26bedcbd06 
It'll build `libclc` in `ci_default` mode and run it through linux build and test. I've decided against adding another target.